src/store/workflows.module.js

```diff
@@ -1,18 +1,108 @@
 // raw GraphQL response data structure
+import omit from 'lodash/omit'
+import { extractGroupState } from '@/utils/tasks'
+
 const state = {
-  workflows: []
+  workflows: [],
+  workflowTree: []
 }
 
 const mutations = {
   SET (state, data) {
     // TODO: when subscriptions are introduced this will have to apply
     // deltas to the store
     state.workflows = data
+  },
+  SET_WORKFLOW_TREE (state, workflowTree) {
+    state.workflowTree = workflowTree
+  }
+}
+
+/**
+ * Compute a map of cycles, where the key is the workflow ID, and the
+ * value is the cyclepoint name.
+ *
+ * @param workflows
+ * @returns {Map<any, any>}
+ * @private
+ */
+function _getCycles (workflows) {
+  const cycles = new Map()
+  for (const workflow of workflows) {
+    if (Object.prototype.hasOwnProperty.call(workflow, 'taskProxies')) {
+      console.log(workflow.taskProxies)
+      if (!cycles.get(workflow.id)) {
+        cycles.set(workflow.id, new Set())
+      }
+      for (const proxy of workflow.taskProxies) {
+        cycles.get(workflow.id).add(proxy.cyclePoint)
+      }
+    }
+  }
+  return cycles
+}
+
+/**
+ * Compute a workflow tree where the root is the workflow node, followed by
+ * the cycle points, and finally by task proxies.
+ *
+ * Every node has a .name property for display.
+ *
+ * @param workflows
+ * @returns {[]}
+ * @private
+ */
+function _getWorkflowTree (workflows) {
+  const cycles = _getCycles(workflows)
+  const workflowTree = []
+  if (cycles.size > 0) {
+    for (const workflow of workflows) {
+      // add workflow minus taskProxies, with children
+      const simplifiedWorkflow = omit(workflow, 'taskProxies')
```

Comment:
I'm guessing this makes a copy of the workflow minus the `taskProxies`.

Can we get away without creating a new object, could we just add a new field to the workflow?

Reply:
Sure, done! :+1: